compiler/rustc_codegen_ssa/src/mir/locals.rs

```diff
@@ -37,16 +37,7 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
     pub(super) fn initialize_locals(&mut self, values: Vec<LocalRef<'tcx, Bx::Value>>) {
         assert!(self.locals.values.is_empty());
 
-        for (local, value) in values.into_iter().enumerate() {
-            match value {
-                LocalRef::Place(_) | LocalRef::UnsizedPlace(_) | LocalRef::PendingOperand => (),
-                LocalRef::Operand(op) => {
-                    let local = mir::Local::from_usize(local);
-                    let expected_ty = self.monomorphize(self.mir.local_decls[local].ty);
-                    assert_eq!(expected_ty, op.layout.ty, "unexpected initial operand type");
-                }
-            }
-
+        for (_, value) in values.into_iter().enumerate() {
```

Comment:
Can you modify this to a FIXME like below? And add a test from the issue that was recently minimized?